lib/drivers/oneTouchUltra2.js

```diff
@@ -0,0 +1,386 @@
+  var DMUpload = DM.concat('\x44\x4d\x50');
+  var DMGlucoseUnits = DM.concat('\x44\x4d\x53\x55\x3f');
+  var DMTimeFormat = DM.concat('\x44\x4d\x53\x54\x3f');
+  var CR = 0x0D;
+  var LF = 0x0A;
+
+  function buildDateTime(date, time) {
+    var fmt = 'MM/DD/YY HH:mm:ss';
+    var ddate = date + ' ' + time;
+    var parsed = sundial.parseFromFormat(ddate, fmt);
+    var dev = sundial.formatDeviceTime(parsed);
+    var utc = sundial.applyTimezone(dev, cfg.timezone).toISOString();
+    var offset = sundial.getOffsetFromZone(utc, cfg.timezone);
+    return { dev: dev, utc: utc, offset: offset };
+  }
+
+  var buildPacket = function (bytes) {
+    var packet = {
+      id: 'OneTouchUltra2 ', // wait for the serial number later
+      bytes: bytes,
+      valid: false,
+      packet_len: 0,
+      lcb: 0,
+      payload: null,
+      crc: 0
+    };
+
+    var packet_len = bytes.byteLength;
+    packet.packet_len = packet_len;
+    packet.valid = true;
+    return packet;
+  };
+
+  var otu2MessageHandler = function (buffer) {
+    if (buffer.len() < 1) {
+      return false;
+    }
+    // there's enough there to try, anyway
+    var packet = buildPacket(buffer.bytes());
+    if (packet.packet_len !== 0) {
+      // remove the now-processed packet
+      buffer.discard(packet.packet_len);
+    }
+
+    if (packet.valid) {
+      return packet;
+    } else {
+      return null;
+    }
+  };
+
+  var otu2ErrorHandler = function(info) {
+    // we don't care too much for timeout errors
+    if(info.error && info.error!=='timeout'){
+      console.log('DEBUG OTU2ERROR:', info);
+    }
+    if (info.connectionId && info.error) {
+      if (info.error == 'timeout') {
+        return -9; // TODO check this
+      }
+    }
+  };
+
+  var byteConcat = function (a, b){
+      var aLength = a.length,
+      result = new Uint8Array(aLength + b.length);
+      result.set(a);
+      result.set(b, aLength);
+      return result;
+  };
+
+  var listenForPacket = function (timeout, callback) {
+    //console.log('DEBUG listenForPacket');
+    var abortTimer = setTimeout(function () {
+      clearInterval(listenTimer);
+      console.log('TIMEOUT');
+      callback('TIMEOUT', null);
+    }, timeout);
+
+    var listenTimer = setInterval(function () {
+      var wholebytes = new Uint8Array([0]), pkt;
+
+      while(serialDevice.hasAvailablePacket()) {
+        pkt = serialDevice.nextPacket();
+        wholebytes = byteConcat(wholebytes, pkt.bytes);
+      }
+      pkt = buildPacket(wholebytes);
+      var tostr = String.fromCharCode.apply(String, pkt.bytes);
+      console.log('DEBUG OTU2 listenForPacket:', pkt, tostr);
+      clearTimeout(abortTimer);
+      clearInterval(listenTimer);
+
+      callback(null, pkt);
+    }, 1000);     // spin on this one not so quickly
+  };
+
+  var probe = function(cb){
+    console.log('attempting probe of oneTouch Ultra2 - don\'t probe anything');
+    //otu2CommandResponse(DM, cb);
```

Comment:
you mean the otu2commandResponse? We realise there was no need (a strong one) to just fire a command to the device on start, since for each command we are sending a DM for collecting data later.